docs/designs/high-level.md

```diff
@@ -1,33 +1,104 @@
+
+The Reconcilers and Components communicate via a Status object of the Custom Resource. It contains a data structure that exposes the current status of each reconciler and component.
+
+The Manager retrieves the current custom resource, based on the ID, from Kubernetes (using its API) and checks to see whether it was deleted. If it was, nothing needs to be done since all Kubernetes objects created by the operator are using [Owner Reference](https://kubernetes.io/blog/2021/05/14/using-finalizers-to-control-deletion/#owner-references) to the custom resource, thus are deleted in a cascading fashion.
+
+If it was created or updated, the Manager uses the retrieved custom resource YAML to parse it and produces a data structure representing it. It then executes a chain of Reconcilers, passing the Custom Resource data structure as main input.
+
+Each Reconciler executes and return two possible return values:
+1. Continue: I'm done, you can continue to the next Reconciler in the chain.
+2. Return-to-start: I've started an action, and changed the k8s objects / OpenSearch Configuration, let's go back and run the chain again to verify if some reconciler need to respond to that. When that value is returned, the Manager restarts the execution of the Reconciler chain
+
+On both accounts the reconciler updates its status in the Status element of the custom resource.
+
+Aside from Reconcilers we also have Components. Those are software objects whose responsibilities don't fall into the category of reconciling Kubernetes objects based on the Custom Resource. The components we have are:
+1. Scaler: in charge of doing graceful scale down of an OpenSearch Node.
+2. Auto-Scaler (Not written yet): Will respond to OpenSearch metrics polled periodically from Prometheus and modify amount Data Nodes and/or Coordinators of the cluster to respond to varying load.
+
+### Monitoring
+
+The Operator deploys Prometheus with built-in dashboards and Alerts, Grafana and Alerts Manager providing the user a batteries-included experience for monitoring OpenSearch. The complete reference is included in the [Monitoring Design](monitoring.md). 
+
+### Security
+
+The operator wants to accomplish two goals from security standpoint: Secure from the get-go and ease of configuring OpenSearch Security Plugin. The complete design for this section is in the [Security Design](security.md)
+
+### Interacting with the Operator
+
+Can be done in 2 ways:
 1. Updating its CRD. 
 2. Sending API request.
```

Comment:
What API? I think currently the only way is via the custom resource

Reply:
I'll delete it.